tests/analytics/live-event-analytics.test.ts

```diff
@@ -0,0 +1,117 @@
+            
+            postSpy = jest.spyOn(sdk.fetcher.fetcher, "post");
+            putSpy = jest.spyOn(sdk.fetcher.fetcher, "put");
+        });
+
+        describe("trackAttendance", () => {
+            let params;
+            
+            beforeEach(() => {
+                params = {
+                    boardId: 1,
+                    widgetId: "abc",
+                    guid: "123",
+                    activityType: "attended_event",
+                    eventName: "My event",
+                };
+            });
+
+            it("checks that board id is incorporated in the URL", async () => {
+                await liveEventAnalytics.trackAttendance("create", params);
+
+                expect(postSpy).toHaveBeenCalledWith(
+                    expect.stringContaining(`/live_board/v1/boards/${params.boardId}/attend_activities`),
+                    expect.any(Object),
+                );
+            });
+
+            it("checks that all parameters are passed to the payload", async () => {
+                await liveEventAnalytics.trackAttendance("create", params);
+
+                expect(postSpy).toHaveBeenCalledWith(
+                    expect.any(String),
+                    expect.objectContaining({
+                        widget_id: params.widgetId,
+                        guid: params.guid,
+                        event_name: params.eventName,
+                        activity_type: expect.any(Number),
+                    }),
+                );
+            });
+
+            it("checks that trackAttendance with create type triggers a post request", async () => {
+                await liveEventAnalytics.trackAttendance("create", params);
+
+                expect(postSpy).toHaveBeenCalled();
+                expect(putSpy).not.toHaveBeenCalled();
+            });
+
+            it("checks that trackAttendance with update type triggers a put request", async () => {
+                await liveEventAnalytics.trackAttendance("update", params);
+
+                expect(putSpy).toHaveBeenCalled();
+                expect(postSpy).not.toHaveBeenCalled();
+            });
+
+            it("checks that activityType attended_event translate to type 0", async () => {
+                params.activityType = "attended_event";
+                await liveEventAnalytics.trackAttendance("create", params);
+
+                expect(postSpy).toHaveBeenCalledWith(expect.any(String), expect.objectContaining({activity_type: 0}));
+            });
+
+            it("checks that activityType viewed_recording translate to type 2", async () => {
+                params.activityType = "viewed_recording";
+                await liveEventAnalytics.trackAttendance("create", params);
+
+                expect(postSpy).toHaveBeenCalledWith(expect.any(String), expect.objectContaining({activity_type: 2}));
+            });
+        });
+    });
+
+    describe("In preview", () => {
+            beforeEach(async () => {
```

Comment:
something weird with indentation

Reply:
It's 4 spaces, which is out standard. I tried formatting with both lint and prettier, and they didn't change anything about the indentation